.github/workflows/k8s-ci.yml

```diff
@@ -0,0 +1,380 @@
+          args: BuildOpaBundle
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Cache Build dependencies
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.workflow }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Build cloudbeat-docker
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: false
+          tags: cloudbeat:latest
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
+          outputs: type=docker,dest=/tmp/cloudbeat-${{ env.CONTAINER_SUFFIX }}.tar
+
+      - name: Build elastic-agent
+        run: |
+          make build_elastic_agent_docker_image
+
+      - name: Build pytest-docker
+        uses: docker/build-push-action@v4
+        with:
+          context: ./tests/.
+          push: false
+          tags: cloudbeat-test:latest
+          cache-from: type=local,mode=max,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+          outputs: type=docker,dest=/tmp/pytest-${{ env.CONTAINER_SUFFIX }}.tar
+
+      - name: Cache docker images
+        uses: actions/cache@v3
+        with:
+          path: /tmp/*.tar
+          key: ${{ runner.os }}-dockers-cache-${{ env.CONTAINER_SUFFIX }}
+
+      - name: Move cache
+        run: |
+          ./.ci/scripts/gh-cache.sh
+        shell: bash
+
+  k8s_functional_tests:
+    # Run only selected tests on PRs
+    if: github.event_name == 'pull_request'
+    name: ${{ matrix.test-target }}-tests
+    needs: [Build]
+    runs-on: ubuntu-20.04
+    timeout-minutes: 55
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - test-target: pre_merge_agent
+            range: ''
+            values_file: tests/deploy/values/ci-sa-agent.yml
+          - test-target: pre_merge
+            range: ''
+            values_file: tests/deploy/values/ci.yml
+          - test-target: file_system_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: k8s_object_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: process_scheduler_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: process_api_server_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: process_controller_manager_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: process_etcd_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
+          - test-target: process_kubelet_rules
+            range: '0..4'
+            values_file: tests/deploy/values/ci.yml
```

Comment:
@oren-zohar, I think using [documentation](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) we can try to refactor this section by the following:
```
  matrix:
    test-target: [file_system_rules, k8s_object_rules, process_scheduler_rules, process_api_server_rules, process_controller_manager_rules, process_etcd_rules, process_kubelet_rules]
    values_file: [tests/deploy/values/ci.yml]
    range: '0..4'
        include:
          - test-target: pre_merge_agent
            range: ''
            values_file: tests/deploy/values/ci-sa-agent.yml
          - test-target: pre_merge
            range: ''
            values_file: tests/deploy/values/ci.yml
```